qiime2/plugin/testing.py

```diff
@@ -103,6 +103,9 @@ def get_data_path(self, filename):
     def get_transformer(self, from_type, to_type):
         """Convenience method for getting a registered transformer.
 
+        This helper deliberately side-steps the framework's validation machinery,
+        that way it is possible for plugin developers to test failing conditions.
```

Comment:
Just a very minor suggestion to improve the flow:  change "that way it is possible" to "making it possible" or "so that it is possible".